Source/DafnyStandardLibraries/src/Std/TargetSpecific/Duration.dfy

```diff
@@ -0,0 +1,405 @@
+include "DateTimeConstant.dfy"
+include "../Collections/SeqExt.dfy"
+
+module Std.Duration {
+  import opened DateTimeConstant
+  import opened Strings
+  import opened Collections.SeqExt
+  import opened Collections.Seq
+  import opened BoundedInts
+  import opened Arithmetic.Power
+
+  datatype Duration = Duration(
+    seconds: uint16,
+    millis: uint16
+  ) {
+
+    ghost predicate Valid() {
+      millis < (MILLISECONDS_PER_SECOND as uint16)
+    }
+  }
+
+  function ToTotalMilliseconds(d: Duration): uint64
+    requires d.Valid()
+  {
+    var product := (d.seconds as uint64) * (MILLISECONDS_PER_SECOND as uint64);
+    var sum := product + (d.millis as uint64);
+    sum
+  }
+
+  function FromMilliseconds(ms: uint32): Duration
+    ensures FromMilliseconds(ms).Valid()
+  {
+    /**
+    Duration((ms / (MILLISECONDS_PER_SECOND as uint32)) as uint16,
+             (ms % (MILLISECONDS_PER_SECOND as uint32)) as uint16)*/
```

Comment:
As a rule don't leave commented-out code in source files. They just fall out of date anyway. Prior versions can always be retrieved from source control (i.e. git) if necessary.